src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1936,31 +1940,55 @@ public virtual bool Accepts(DType type, out KeyValuePair<string, DType> schemaDi
                     accepts = type.Kind == DKind.Deferred || type.Kind == DKind.Unknown;
                     break;
                 case DKind.String:
-                    accepts =
-                        type.Kind == Kind ||
-                        type.Kind == DKind.Hyperlink ||
-                        type.Kind == DKind.Image ||
-                        type.Kind == DKind.PenImage ||
-                        type.Kind == DKind.Media ||
-                        type.Kind == DKind.Blob ||
-                        type.Kind == DKind.Unknown ||
-                        type.Kind == DKind.Deferred ||
-                        type.Kind == DKind.Guid ||
-                        (type.Kind == DKind.Enum && Accepts(type.GetEnumSupertype()));
+                    if (usePowerFxV1CompatibilityRules)
+                    {
+                        accepts =
+                            type.Kind == Kind ||
+                            type.Kind == DKind.Unknown ||
+                            type.Kind == DKind.Deferred ||
+                            (type.Kind == DKind.Enum && Accepts(type.GetEnumSupertype(), exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true));
+                    }
+                    else
+                    {
+                        accepts =
+                            type.Kind == Kind ||
+                            type.Kind == DKind.Hyperlink ||
+                            type.Kind == DKind.Image ||
+                            type.Kind == DKind.PenImage ||
+                            type.Kind == DKind.Media ||
+                            type.Kind == DKind.Blob ||
+                            type.Kind == DKind.Unknown ||
+                            type.Kind == DKind.Deferred ||
+                            type.Kind == DKind.Guid ||
+                            (type.Kind == DKind.Enum && Accepts(type.GetEnumSupertype(), exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: false));
```

Comment:
>false [](http://example.com/codeflow?start=167&length=5)

usePowerFxV1CompatibilityRules

Reply:
This is inside `if(usePowerFxV1CompatibilityRules)` so I know that the value is true... But I can change to use the variable to be consistent with other places.